tests/regression/58-base-mm-tid/24-phases-sound.c

```diff
@@ -0,0 +1,46 @@
+// PARAM: --set ana.path_sens[+] threadflag --set ana.base.privatization mutex-meet-tid --enable ana.int.interval --set ana.activated[+] threadJoins --set ana.activated[+] thread
+// Tests soundness when additionally thread analysis is enabled, that is able to go back to single-threaded mode after all created joins have been joined.
+#include <pthread.h>
+#include <goblint.h>
+
+int g = 10;
+
+pthread_mutex_t A = PTHREAD_MUTEX_INITIALIZER;
+
+void *t_benign(void *arg) {
+  pthread_mutex_lock(&A);
+  g = 10;
+  __goblint_check(g == 10); //TODO
```

Comment:
What's the status for these TODOs? Are they related to this single-threadedness stuff or just orthogonal?